### DISCUSSION TITLE
Allow sites to serve their own dynamic-theme-fixes.config and inversion-fixes.config

### DISCUSSION THREAD

**Discussion by hjr265** [darkreader/darkreader#11450] [Ideas]
I was wondering if it is possible for Dark Reader to allow sites to serve their own fixes configuration file. 

Sites could add a meta tag like `<meta name="darkreader-dynamic-theme-fixes-config-url" content="...">`. Dark Reader could fetch the file and use it on top of its own `dynamic-theme-fixes.config` file. 

Dark Reader could rely on the file's cache headers to determine how frequently to refresh the file.

I tried to look for such an option here in this GitHub project. But couldn't find one. It seems the only approach available now is to send PRs for my site.

**Comment by bershanskiy** [comment]
@hjr265 Dark Reader currently does not support custom fixes. I'm open to adding this as a feature, after discussing API ergonomics and security implications. As an alternative, right now you could add Dark Reader lock and then manually import Dark Reader module (e.g., from npm) and specify fixes in `DarkReader.enable()`. Would that work for you?

**Reply by BenceSzalai** [reply]
Hi! I am looking exactly for this, but I cannot see an option (key) to provide my fixes string to `DarkReader.enable()`. Is this a feature available, or was it just a theoretical question and it is not really possible to provide fixes to a DarkReader instance loaded using JS?

I assume all valid keys are listed here: https://github.com/darkreader/darkreader/blob/main/src/defaults.ts#L22

UPDATE: nevermind, I've found it, it goes in a second parameter. https://github.com/darkreader/darkreader/blob/main/src/api/index.ts#L21

**Comment by hjr265** [comment]
@bershanskiy That, I think, would work. At least technically.

A bit of a context is that we didn't want to add **native** support for dark mode to the site just yet. At least not until we figured out some branding details around it.

Until then we wanted to encourage the site's users to get an extension for it (in this case Dark Reader). Dark Reader works for us like 85%. But to get the rest 15% in place some fixes are necessary.

If sites can host/serve their own fixes, IMO it could solve, at least somewhat, some of the challenges around maintaining the fixes file within this repository.
